pkg/kubelet/volumes_util.go

```diff
@@ -0,0 +1,79 @@
+package kubelet
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/golang/glog"
+)
+
+// AddMountErrorHint performs some basic analysis
+// on the current mount error returned and will
+// add a user hint or resolution tip for enhanced UXP
+func (kl *Kubelet) AddMountErrorHint(volpath string, volname string, inerr error) error{
```

Comment:
For the record I can totally understand the desire to keep this orthogonal from the volume plugins, but I think it's a fine start (and likely to work better in the Kubelet we have today) to keep the logic about possible error causes at the sites where they occur.  I think that is the simplest way to start, and maybe a better pattern will become evident.